### PR TITLE
Makefile (Thomas version) with everything re-run

### PR DESCRIPTION
**Pull request by thomassiu**
This is my version... feel free to review first

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+data/processed/preprocessor.p data/processed/models.p: src/pre_process_data.py
+	python src/pre_process_data.py --out_path=data/processed/
+
+# Usage: Make results/pipe_best.p
+# Fit and tune model
+results/pipe_best.p \
+results/models_results_cv.png \
+results/best_LR_model.png \
+results/confusion_matrix.png \
+esults/classification_report.png: src/modelling.py \
```
Comment (RamiroMejia) [inline Makefile]: seems like an 'r' is missing here

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+results/confusion_matrix.png \
+results/classification_report.png
+	Rscript -e "rmarkdown::render('doc/vancouver_crime_predict_report.Rmd')"
+
+# Usage: Make clean
+# remove all necessary files for a fresh start of analysis
+clean: 
+	rm -rf data/processed
+	rm -rf data/raw/crimedata_csv_all_years.csv
+	rm -rf results
```
Comment (RamiroMejia) [inline Makefile:80]: i did this :

```
clean: 
	rm -rf data/raw/crimedata_csv_all_years.csv
    rm -rf data/processed/training_df.csv
    rm -rf data/processed/training_feature.csv 
    rm -rf data/processed/training_target.csv 
    rm -rf data/processed/test_feature.csv 
    rm -rf data/processed/test_target.csv 
    rm -rf src/figure-preprocess/data_before_balance.png 
    rm -rf src/figure-preprocess/data_after_balance.png 
    rm -rf src/figure-eda/neighbour_crimes.png 
    rm -rf src/figure-eda/crime_type.png 
    rm -rf src/figure-eda/crime_evolution.png 
    rm -rf src/figure-eda/crime_correlation.png 
    rm -rf src/figure-eda/crime_top5.png
    rm -rf src/crime_vancouver_eda.md
    rm -rf data/processed/preprocessor.p 
    rm- rf data/processed/models.p
    rm - rf results/models_results_cv.png 
    rm - rf results/best_LR_model.png 
    rm - rf results/pipe_best.p 
    rm - rf results/classification_report.png
    rm -rf doc/vancouver_crime_predict_report.md doc/vancouver_crime_predict_report.html
```
Do you think is better to include each out put or delete group of files?

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	rm -rf src/figure-eda src/figure-preprocess
+	rm -rf src/crime_vancouver_eda.md src/crime_vancouver_eda.html	
+	rm -rf doc/vancouver_crime_predict_report.md doc/vancouver_crime_predict_report.html
```
Comment (RamiroMejia) [inline Makefile:83]: I think I messed up in the modelling, I did not think about the dependencies!!! My makefile is very dirty and hard to debug. I checked other groups file and it is cleaner :(

**Review by RamiroMejia — CHANGES_REQUESTED** [review]

**Review by RamiroMejia — APPROVED** [review]
Good job

File: Makefile (hunk 1 above)
Comment (thomassiu) [reply Makefile]: this is fixed now

File: Makefile (hunk 3 above)
Comment (thomassiu) [reply Makefile:83]: I guess it is still fine! One thing we could do (next time) is to segregate the training of model (e.g.  hyperparameters) and fitting the test data.

**Review by RamiroMejia — APPROVED** [review]

**Review by sy25wang — APPROVED** [review]

**Pull request merged**